skaffold/src/main/kotlin/com/google/container/tools/skaffold/run/SkaffoldCommandLineState.kt

```diff
@@ -60,6 +64,24 @@ class SkaffoldCommandLineState(
             throw ExecutionException(message("skaffold.no.file.selected.error"))
         }
 
+        //call skaffold version --- look at createProcess function, the command list lets you know if skaffold isn't on the system by process of elimination
+        if (Paths.get("skaffold").exists()) {
```

Comment:
I think a better way to check that Skaffold is on the PATH is to check the system path directly and see if `skaffold` is there. Its generally advised not to use exception control flow in this way. Something like this (with comments added for clarity):

```
fun isExecutableInPath(executable: String): Boolean {
   val path = System.getenv("PATH") // get the full path string
   return path.split(File.pathSeparator) // split the path string by the path separator
      .asSequence() // use sequence for lazy evaluation
      .map { it + File.separator + executable } // convert each to a possible path to the executable
      .any {
         File(it).exists() // does the file exist?
      }
}
```

then calling `isExecutableInPath("skaffold")` should do the check

Reply:
@ivanporty had a great point that this doesn't check that the file is executable. I would suggest ammending the last line:

`File(it).exists() && File(it).canExecute()`